lib/taiko.js

```diff
@@ -277,7 +277,7 @@ module.exports.goto = async (url, options = { timeout: 30000 }) => {
     if (options.headers) await network.setExtraHTTPHeaders({ headers: options.headers });
     const res = await page.navigate({ url: url });
     if (res.errorText) throw new Error(`Navigation to url ${url} failed.\n REASON: ${res.errorText}`);
-    await waitForNavigation(options.timeout, promises).catch(handleTimeout(options.timeout));
+    if (options.timeout) await waitForNavigation(options.timeout, promises).catch(handleTimeout(options.timeout));
```

Comment:
the problem here is when users provides custom options it is overriding the default option `{timeout: 30000}` , better fix will be to append the options then overriding it